documents-webapp/src/main/java/org/exoplatform/documents/portlet/DocumentGadgetPortlet.java

```diff
@@ -0,0 +1,113 @@
+import org.exoplatform.services.security.Identity;
+import org.exoplatform.social.core.space.SpaceUtils;
+import org.exoplatform.social.core.space.model.Space;
+
+import io.meeds.social.space.plugin.SpaceAclPlugin;
+
+public class DocumentGadgetPortlet extends CMSPortlet {
+
+  private static final String OBJECT_TYPE    = "documentGadget";
+
+  private static final String APPLICATION_ID = "applicationId";
+
+  private static final String CAN_EDIT       = "canEdit";
+
+  private final SecureRandom  random         = new SecureRandom();
+
+  private UserACL             userAcl;
+
+  @Override
+  public void init(PortletConfig config) throws PortletException {
+    super.init(config);
+    this.contentType = OBJECT_TYPE;
+  }
+
+  @Override
+  public void doView(RenderRequest request, RenderResponse response) throws PortletException, IOException {
+    boolean canModifySettings = canModifySettings();
+    request.setAttribute(CAN_EDIT, canModifySettings);
+    request.setAttribute(APPLICATION_ID, getOrCreateApplicationId(request.getPreferences()));
+    super.doView(request, response);
+  }
+
+  @Override
+  public void processAction(ActionRequest request, ActionResponse response) throws IOException, PortletException {
+    if (!canModifySettings()) {
+      throw new PortletException("User is not allowed to edit settings");
+    }
+    PortletPreferences preferences = request.getPreferences();
+    Enumeration<String> parameterNames = request.getParameterNames();
+    while (parameterNames.hasMoreElements()) {
+      String name = parameterNames.nextElement();
+      if (StringUtils.equals(name, "action") || StringUtils.contains(name, "portal:")) {
+        continue;
+      }
+      String value = request.getParameter(name);
+      preferences.setValue(name, value);
+    }
+    preferences.store();
+  }
+
+  private Identity getCurrentIdentity() {
+    return ConversationState.getCurrent() == null ? null : ConversationState.getCurrent().getIdentity();
+  }
+
+  private UserACL getUserAcl() {
+    if (userAcl == null) {
+      userAcl = ExoContainerContext.getService(UserACL.class);
+    }
+    return userAcl;
+  }
+
+  private boolean canModifySettings() {
+    Space space = SpaceUtils.getSpaceByContext();
+    Identity identity = getCurrentIdentity();
+    if (identity == null || getUserAcl().isAnonymousUser(identity)) {
+      return false;
+    } else if (space == null) {
+      return getUserAcl().isAdministrator(identity);
+    } else {
+      return getUserAcl().isAdministrator(identity)
+          || getUserAcl().hasEditPermission(SpaceAclPlugin.OBJECT_TYPE, space.getId(), identity);
+    }
+  }
+
+  private String getOrCreateApplicationId(PortletPreferences preferences) {
+    String applicationId = preferences.getValue(APPLICATION_ID, null);
+    if (applicationId == null) {
+      applicationId = String.valueOf(random.nextLong() & Long.MAX_VALUE);
+      savePreference(APPLICATION_ID, applicationId);
+    }
+    return applicationId;
+  }
```

Comment:
As discussed, need to rework this portlet to rely on `Settings Name` of the portlet rather than redefining this mechanism.

```suggestion
```